lib/src/verifiers/utilities/json-schema-verifier.spec.ts

```diff
@@ -0,0 +1,603 @@
+      });
+    });
+
+    describe("string schema", () => {
+      it("validates", () => {
+        const dataType: StringType = {
+          kind: TypeKind.STRING
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates", () => {
+        const dataType: StringType = {
+          kind: TypeKind.STRING
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+
+      it("validates literal", () => {
+        const dataType: StringLiteral = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates literal", () => {
+        const dataType: StringLiteral = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "not hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+    });
+
+    describe("number schema", () => {
+      it("validates", () => {
+        const dataType: NumberType = {
+          kind: TypeKind.NUMBER
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates", () => {
+        const dataType: NumberType = {
+          kind: TypeKind.NUMBER
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+
+      it("validates literal", () => {
+        const dataType: NumberLiteral = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates literal", () => {
+        const dataType: NumberLiteral = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 53
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+    });
+
+    describe("array schema", () => {
+      it("validates", () => {
```

Comment:
I'd also like to see that it doesn't validate `undefined` and it does validate `[]`.

Reply:
We can test for `[]` but there's no way to test `undefined`. The Array itself is being validated - `undefined` is not an array, so this situation should never happen.